crates/polars-expr/src/reduce/min_max.rs

```diff
@@ -658,3 +660,76 @@ impl<T: PolarsCategoricalType> Reducer for CatMaxReducer<T> {
         }
     }
 }
+
+#[derive(Default)]
+pub struct NullGroupedReduction {
+    length: usize,
+}
+
+impl GroupedReduction for NullGroupedReduction {
+    fn new_empty(&self) -> Box<dyn GroupedReduction> {
+        Box::new(Self::default())
+    }
+
+    fn reserve(&mut self, additional: usize) {
+        self.length += additional;
+    }
+
+    fn resize(&mut self, num_groups: IdxSize) {
+        self.length = num_groups as usize;
+    }
+
+    fn update_group(
+        &mut self,
+        values: &Column,
+        _group_idx: IdxSize,
+        _seq_id: u64,
+    ) -> PolarsResult<()> {
+        assert!(values.dtype() == &DataType::Null);
+
+        // no-op
+        Ok(())
+    }
+
+    unsafe fn update_groups_while_evicting(
```

Comment:
I think I was wrong about my initial advice of not needing to handle evictions. I think you should add a second struct member `num_evictions: usize`, and then here

```rust
for g in group_idxs {
    self.num_evictions += g.should_evict() as usize;
}
```